From fbdc257a6ca616a92bc0c86653a5c34e437e8bf2 Mon Sep 17 00:00:00 2001
From: Steve Wainstead <wainstead@gmail.com>
Date: Wed, 9 Feb 2011 05:00:01 +0000
Subject: [PATCH 135/223] Now the user can specify --enable-expat=[yes|no] to
 ./configure, or leave it out entirely. Default is
 to compile with Expat.

If the expat library is not found, or the expat.h header file is not
found, it still compiles without XML. It should probably fail during
./configure if the user specified --enable-expat.

I may change to a different macro and use --with-xml instead of
--enable-expat.
---
 server/configure.in |   31 +++++++++++++++++++++++--------
 1 file changed, 23 insertions(+), 8 deletions(-)

diff --git a/server/configure.in b/server/configure.in
index 40d43c1..bce4656 100644
--- a/server/configure.in
+++ b/server/configure.in
@@ -317,9 +317,6 @@ AC_SIZE_T
 AC_STRUCT_TM
 AC_TIMEZONE
 
-have_expat=no
-AC_SEARCH_LIBS([XML_ParserCreate], [expat], [have_expat=yes])
-
 MOO_HAVE_FUNC_LIBS(sqrt, -lm /lib/libm.a "-static -lm")
 MOO_HAVE_FUNC_LIBS(mkfifo waitpid sigemptyset, -lposix /lib/libposix.a)
 MOO_HAVE_FUNC_LIBS(accept, "-lsocket -lnsl" -lsocket -linet)
@@ -522,18 +519,36 @@ yes
 
 AC_CHECK_HEADERS([arpa/inet.h fcntl.h float.h limits.h malloc.h memory.h netdb.h netinet/in.h stdlib.h string.h sys/ioctl.h sys/param.h sys/socket.h sys/time.h unistd.h])
 
-if test "x${have_expat}" = xyes; then
-    AC_CHECK_HEADERS([expat.h], [], [have_expat=no])
-fi
 
-if test "x${have_expat}" = xno; then
-    AC_MSG_WARN([
+# Expat support: search for the library, then expat.h.
+# Let the user choose to enable expat or not
+AC_ARG_ENABLE([expat],
+        [  --enable-expat	enable XML parsing via the Expat library],
+        [expat=${enableval}],
+        [expat=yes])
+
+if test "x${expat}" = xyes; then
+
+   have_expat=no
+   AC_SEARCH_LIBS([XML_ParserCreate], [expat], [have_expat=yes])
+
+   if test "x${have_expat}" = xyes; then
+
+      AC_CHECK_HEADERS([expat.h], [], [have_expat=no])
+
+      fi
+
+      if test "x${have_expat}" = xno; then
+      AC_MSG_WARN([
   ----------------------------------------------------
   Unable to find the Expat XML library on this system.
   Building a moo server without XML support.
   ----------------------------------------------------])
+  fi
 fi
 
+
+
 AC_C_INLINE
 AC_FUNC_ALLOCA
 AC_FUNC_ERROR_AT_LINE
-- 
1.7.9.5

